### PR TITLE
nfcStartOtherTagScanning(): Call StartCommunication() for Old3DS compatibility.

### PR DESCRIPTION
**Pull request by HiddenRambler**
Duplicated Old 3DS compatibility code from nfcStartTagScanning() for nfcStartOtherTagScanning(). Could refactor more if needed.

### CODE REVIEW

**Comment by fincs** [comment]
Instead of duplicating the code, why not extract it out to a separate static function?

**Comment by HiddenRambler** [comment]
Refactored the shared code out.

File: libctru/source/services/nfc.c [hunk 1]
```diff
@@ -15,12 +15,14 @@ static NFC_OpType nfc_optype = NFC_OpType_NFCTag;
 static Result NFC_Initialize(NFC_OpType type);
 static Result NFC_Shutdown(NFC_OpType type);
 
+static Result NFC_StartCommsAdapter();
```
Comment (fincs) [inline libctru/source/services/nfc.c]: Should be `(void)` just like the others - in C, `()` doesn't indicate an empty parameter list.

File: libctru/source/services/nfc.c [hunk 2]
```diff
@@ -63,49 +65,62 @@ Handle nfcGetSessionHandle(void)
 	return nfcHandle;
 }
 
-Result nfcStartScanning(u16 inval)
+static Result NFC_StartCommsAdapter()
```
Comment (fincs) [inline libctru/source/services/nfc.c]: Same thing here

**Comment by HiddenRambler** [comment]
Done.

**Comment by WinterMute** [comment]
What's the difference between StartOtherTagScanning and StartTagScanning here? From a brief glance it looks to me like the former is a superset of the latter and perhaps they should be only differentiated internally?

Would it be reasonable to provide a single interface which subsequently calls the appropriate function based on it's parameters? 

@yellows8

**Comment by HiddenRambler** [comment]
This would depend on how much the ctrulib will abstract away the native api.

While we are at it. According to my trial and error investigation the nfcCmd21() and nfcCmd22() seems to enable and disable keeping the NFC tag powered up during commands. This is usefull to perform as part of a single session of the tag. see: https://github.com/HiddenRambler/Thenaya/blob/master/source/nfc.c#L332

perhaps we can rename these methods to more sensible names?

**Comment by WinterMute** [comment]
Merging this in as is for next release.

Abstracting out the NFC api more can only be a good thing I think and better names for some of the partially investigated functions would be appreciated.

We can look at that for another PR though rather than leaving bugfixes sitting in review.

**Comment by yellows8** [comment]
"What's the difference between StartOtherTagScanning and StartTagScanning here?" Basically enables using this: https://3dbrew.org/wiki/NFC:SendTagCommand

**Comment by WinterMute** [comment]
@yellows8 ok, is there any reason why we would want to use the more limited version in preference?

**Comment by yellows8** [comment]
Whenever one wants to send raw NFC commands.

**Pull request merged**